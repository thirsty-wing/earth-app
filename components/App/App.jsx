import { SafeAreaProvider } from "react-native-safe-area-context";
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { ApolloClient, ApolloProvider, InMemoryCache } from "@apollo/client";

import HomeScreen from "../HomeScreen";
import ContinentScreen from "../ContinentScreen";

const client = new ApolloClient({
  uri: "https://countries.trevorblades.com/graphql",
  cache: new InMemoryCache(),
});

const Stack = createNativeStackNavigator();

const linking = {
  config: {
    screens: {
      Home: "",
      Continent: "continent/:code",
    },
  },
  prefixes: [],
};

export function App() {
  return (
    <SafeAreaProvider>
      <ApolloProvider client={client}>
        <NavigationContainer linking={linking}>
          <Stack.Navigator screenOptions={{ headerShown: false }}>
            <Stack.Screen name="Home" component={HomeScreen}/>
            <Stack.Screen name="Continent" component={ContinentScreen}/>
          </Stack.Navigator>
        </NavigationContainer>
      </ApolloProvider>
    </SafeAreaProvider>
  );
}

export default App;
