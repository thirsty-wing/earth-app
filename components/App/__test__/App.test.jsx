import App from "../App";
import { shallow } from "../../../utils/test/shallow";


jest.mock("@react-navigation/native-stack", () => ({
  createNativeStackNavigator: () => ({
    Navigator: jest.fn(),
    Screen: jest.fn(),
  }),
}));


it("should render", () => {
  const content = shallow(<App/>);

  expect(content).toMatchSnapshot();
});

