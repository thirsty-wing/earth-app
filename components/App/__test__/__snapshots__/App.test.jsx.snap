// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<SafeAreaProvider>
  <mockConstructor
    client={mockConstructor {}}
  >
    <ForwardRef(NavigationContainerInner)
      linking={
        {
          "config": {
            "screens": {
              "Continent": "continent/:code",
              "Home": "",
            },
          },
          "prefixes": [],
        }
      }
    >
      <mockConstructor
        screenOptions={
          {
            "animation": "default",
            "headerShown": false,
          }
        }
      >
        <mockConstructor
          component={[Function]}
          name="Home"
        />
        <mockConstructor
          component={[Function]}
          name="Continent"
        />
      </mockConstructor>
    </ForwardRef(NavigationContainerInner)>
  </mockConstructor>
</SafeAreaProvider>
`;
